### DISCUSSION TITLE
cx.read_sql almost 2.5x slower than pandas.read_sql

### DISCUSSION THREAD

**Discussion by gsakkis** [sfu-db/connector-x#635] [General]
I was excited to come across connector-x for improving some queries but for some reason the performance is nowhere close to the benchmarks shown at the README. For example for the same query:
- pandas: 22s, mem usage 650MB
- connector-x-: 52s, mem usage 500MB

Memory usage was computed from `psutil.virtual_memory().used`

This is against a MS SQL Server with pyodb 5.1.0 and ODBC Driver 18 for SQL Server. Any idea why it's 2.5x slower than pandas?

**Comment by wangxiaoying** [comment]
Hi @gsakkis , to answer your question we may need more information about your workload. But in general you can check on this discussion [here](https://github.com/sfu-db/connector-x/discussions/194#discussioncomment-1800031).
